### PR TITLE
Add support for multiple root stores

### PR DESCRIPTION
**Pull request by dadrian**

### CODE REVIEW

File: certificate.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+		setStatus: func(v *zsearch.CertificateValidation, s *zsearch.RootStoreStatus) { v.Microsoft = s },
+	},
+	{
+		verifier:  &verifier.GoogleCTPrimary,
+		getStatus: func(v *zsearch.CertificateValidation) *zsearch.RootStoreStatus { return v.GetGoogleCtPrimary() },
+		setStatus: func(v *zsearch.CertificateValidation, s *zsearch.RootStoreStatus) { v.GoogleCtPrimary = s },
+	},
+}
+
+func ProcessCertificate(in <-chan []byte, out chan<- []byte, wg *sync.WaitGroup, procWG *sync.WaitGroup) {
```
Comment (zakird) [inline certificate.go]: This should probably be named more aptly given that it doesn't just process a single certificate, it consumes things on a channel.

File: certificate.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+			for _, s := range storesToVerify {
+				old := s.getStatus(validation)
+				new := generateRootStoreStatus(parsed, s.verifier, old, presentedChain, now)
+				s.setStatus(validation, new)
+				for _, parent := range new.Parents {
+					parents[string(parent)] = empty{}
+				}
+			}
+
+			// Assume the parents listed on the certificate are wrong.
```
Comment (zakird) [inline certificate.go]: Can this be pulled at this point? We've run everything through once.

File: certificate.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+			//ZLint
+			ZLint := runZlintProtobuf(parsed)
+			c.Zlint = &ZLint
+		}
+		c.ParseVersion = parseVersion
+		c.PostProcessed = true
+		c.PostProcessTimestamp = uint32(now.UTC().Unix())
+		// there are some old bad historical records without fingerprints
+		if len(recvd.Record.Sha256Fp) == 0 {
+			log.Fatal("empty anonymous record fingerprint")
```
Comment (zakird) [inline certificate.go]: Woah, does this really need a fatal?

File: certificate.go (hunk 2 above)
Comment (dadrian) [inline certificate.go]: We need to fix ztag to stop setting parents, if we haven't already.

File: certificate.go (hunk 3 above)
Comment (dadrian) [inline certificate.go]: It was fatal before!

File: certificate.go (hunk 2 above)
Comment (dadrian) [inline certificate.go]: Also, this won't make a difference until a root store changes.

**Pull request merged**